sequencer/validator.go

```diff
@@ -0,0 +1,26 @@
+package sequencer
+
+import "github.com/sig-0/go-ibft/message"
+
+// Validator represents a unique consensus actor in the network.
+// Its role in the protocol is to broadcast (signed) consensus messages and make proposals (if elected)
+type Validator interface {
+	message.Signer
+
+	// BuildProposal returns this validator's proposal for given sequence
+	BuildProposal(sequence uint64) []byte
+
+	// IsValidProposal checks if the provided proposal is valid for given sequence
+	IsValidProposal(proposal []byte, sequence uint64) bool
+}
+
+type ValidatorSet interface {
```

Comment:
You are mixing frogs and grandmothers my friend :)

Reply:
Why would the validator set need to answer the question "who is the proposer" or "is there quorum"?

Reply:
I'd call this interface `Verifier`, or something similar